snippets/uart_bt/README.rst

```diff
@@ -0,0 +1,37 @@
+.. _snippet-uart_bt:
+
+UART over Bluetooth LE Snippet (uart_bt)
+########################################
+
+.. code-block:: console
+
+   west build -S uart_bt [...]
+
+Overview
+********
+
+This snippet redirects serial console and serial shell output to a UART over BLE instance.
+The Bluetooth Serial device used shall be configured using :ref:`devicetree`.
```

Comment:
This is a duplicate of MCUmgr shell, so the point from the previous discussion stands to discuss why we need a duplicate of this in-tree

Reply:
As you said yourself, it's really not a duplicate:
MCUmgr is a binary protocol, with compression, error checking and whole app ecosystem.
NUS is a best-effort data pipe, just like standard UART.

Now obviously MCUmgr is the robust solution here. 
But there is an argument to be made as why we still have serial backends in the zephyr shell, and why "dumb" serial is still used for e.g. the linux boot console: *it's simple*.

There are just less moving pieces and less code overall, so less things that can go wrong or that have to be debugged. This is my argument as to why this "inferior" BLE backend should be allowed in-tree.

Reply:
If this is using a standard bluetooth UART service then that would be wonderful, please link to the profile if so. If not, then it's irrelevant because a vendor service is a vendor service, which is custom and non-standard.

Reply:
Bluetooth WG:

First of all, NUS has become over the years a de-facto standard for UART over BLE. This can easily be demonstrated by looking at the sheer number of NUS-compatible client and server implementations in the wild:

https://thejeshgn.com/2016/10/01/uart-over-bluetooth-low-energy/
https://github.com/basilfx/go-ble-nordic-uart
https://www.esp32.com/viewtopic.php?t=2147

Note that there was an attempt to standardize UART over BLE that was unsuccessful, so it is very unlikely, unless Nordic pushes NUS through the spec process, that there will ever be a standard for this.

Furthermore, the overlap between this PR and the functionality provided by mcumgr is minimal: only the actual use of the NUS UART implementation for transporting the console overlaps, the rest is different. Notably, this PR implements the Zephyr UART, which then as a consequence makes it possible to use that UART as a console transport.

I see no issue at all in this particular case to support both transports for the console. They are very different in implementation, the client-side requires different software packages, and perhaps most importantly, users want to be able to use NUS in Zephyr both as a regular UART as well as a console transport.

Reply:
> First of all, NUS has become over the years a de-facto standard for UART over BLE

Well this is simply not true, it might be used by many projects just because "it is there" but that does not mean anything about standardisation. No different than with e.g. wifi routers use pre-published specs that then are incompatible with the published specification and incompatible with any other device.

Anyhow, requesting for this to be brought up at and discussed at architecture working group

Reply:
Alright, I will bring it up tomorrow